src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,299 @@
+        /// </summary>
+        public byte[] SegmentOrder
+        {
+            get { return _segmentOrder; }
+            set
+            {
+                if (value.Length != MaxSegments)
+                    throw new ArgumentException($"Size of {nameof(SegmentOrder)} can only be 6 length");
+                // Check if we have all values from 0 to 5
+                bool allExist = true;
+                for (int i = 0; i < MaxSegments; i++)
+                {
+                    allExist &= Array.Exists(value, e => e == i);
+                }
+                if (!allExist)
+                    throw new ArgumentException($"{nameof(SegmentOrder)} needs to have all existing segments from 0 to 5");
+                value.CopyTo(_segmentOrder, 0);
+            }
+        }
+
+        /// <summary>
+        /// Set the screen on or off
+        /// </summary>
+        public bool ScreenOn
+        {
+            get { return _screenOn; }
+
+            set
+            {
+                _screenOn = value;
+                DisplayRaw(0, _lastDisplay[0]);
+            }
+        }
+
+        /// <summary>
+        /// Adjust the screen brightness from 0 to 7
+        /// </summary>
+        public byte Brightness
+        {
+            get { return _brightness; }
+            set
+            {
+                if (value > 7)
+                    throw new ArgumentException($"{nameof(Brightness)} can't be more than 7");
+                _brightness = value;
+                DisplayRaw(0, _lastDisplay[0]);
+            }
+        }
+
+        private PinValue WriteByte(byte data)
+        {
+            // We send data by 8 bits
+            for (byte i = 0; i < 8; i++)
+            {
+
+                _controller.Write(_pinClk, PinValue.Low);
+                DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+                // LSB first
+                if ((data & 0x01) == 0x01)
+                    _controller.Write(_pinDio, PinValue.High);
+                else
+                    _controller.Write(_pinDio, PinValue.Low);
+                // LSB first
+                data >>= 1;
+                _controller.Write(_pinClk, PinValue.High);
+                DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            }
+
+            // Wait for the acknoledge
+            _controller.Write(_pinClk, PinValue.Low);
+            _controller.Write(_pinDio, PinValue.High);
+            _controller.Write(_pinClk, PinValue.High);
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+            _controller.SetPinMode(_pinDio, PinMode.Input);
+
+            // Wait 1 µs, it's the waiting time between clk up and down
+            // That's according to the documentation
+            DelayHelper.DelayMicroseconds(ClockWidthMicroseconds, true);
+
+            var ack = _controller.Read(_pinDio);
+            if (ack == PinValue.Low)
```

Comment:
Can you add a comment of what is going on here? If we get a PinValue.High does that mean we didn't get an acknowledge from the device?

Reply:
Yes, that is what is happening. So concretely, you will miss a character or have a character shift. 
The device can act as a keyboard reader as well. This feature is mainly used when you are trying to read the keys. I did not implement this feature as it's mostly not used in hardware setup I found.